src/mlpack/methods/preprocess/image_converter_main.cpp

```diff
@@ -0,0 +1,139 @@
+/**
+ * @file image_converter_main.cpp
+ * @author Jeffin Sam
+ *
+ * A CLI executable to load and save a image dataset.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include <mlpack/core/util/cli.hpp>
+#include <mlpack/core.hpp>
+
+using namespace mlpack;
+using namespace mlpack::util;
+using namespace arma;
+using namespace std;
+using namespace mlpack::data;
+
+
+PROGRAM_INFO("Image Converter",
+    // Short description.
+    "A utility to load an image or set of images into a single dataset that"
+    "can then be used by other mlpack methods and utilities. This can also"
+    "unpack an image dataset into individual files.",
+    // Long description.
+    "This utility takes a image or an array of images and loads them to a"
+    " matrix. You can specify the height " + PRINT_PARAM_STRING("height") +
+    " width " + PRINT_PARAM_STRING("width") + " and channel " +
+    PRINT_PARAM_STRING("channel") + " of the images that needs to be loaded. "
```

Comment:
Doesn't STB automatically detect these from the image, though, if they aren't specified?  If that's the case we should probably point that out in the documentation here too.

Reply:
Yes, We should point that in documentation

Reply:
Did you want to add that to the documentation here?  I might suggest something like

```
You can optionally specify the height ... ... of the images that need to be loaded;
otherwise, these parameters will be automatically detected from the image.
```

Reply:
It might also be nice to figure out what happens if we specify an incorrect size.  Might need to check with STB for that... does it just crop the image or pad it?